src/Indigo/Contracts/HomebaseLite/Impl/Metadata.hs

```diff
@@ -0,0 +1,90 @@
+-- SPDX-FileCopyrightText: 2022 Tezos Commons
+-- SPDX-License-Identifier: LicenseRef-MIT-TC
+
+{-# LANGUAGE OverloadedLists #-}
+
+module Indigo.Contracts.HomebaseLite.Impl.Metadata
+  ( metadataBigMap
+  , versionString
+  , gitRev
+  ) where
+
+import Indigo hiding (description, name, (<>))
+
+import Data.Aeson (encode)
+import qualified Data.ByteString.Lazy as BS
+import Data.Text.Internal.Builder (toLazyText)
+import Language.Haskell.TH.Lib
+import Language.Haskell.TH.Syntax
+
+import qualified Lorentz.Contracts.Spec.TZIP16Interface as TZ16
+import Morley.Metadata (compileViewCodeTH, mkSimpleMichelsonStorageView)
+import Morley.Micheline (toExpression)
+import Morley.Michelson.Text (unsafeMkMText)
+import Morley.Util.Markdown (HeaderLevel(..))
+
+import Indigo.Contracts.HomebaseLite.Impl.Admin ()
+import Indigo.Contracts.HomebaseLite.Impl.Metadata.Views
+import Indigo.Contracts.HomebaseLite.Impl.Utils
+import Indigo.Contracts.HomebaseLite.Impl.Vote ()
+import Indigo.Contracts.HomebaseLite.Types
+
+metadataBigMap :: MetadataConfig -> TZ16.MetadataMap BigMap
+metadataBigMap conf = TZ16.metadataURI (TZ16.tezosStorageUri TZ16.selfHost metadataKey)
+  <> [(metadataKey, BS.toStrict $ encode $ metadataJSON conf)]
+  where metadataKey = [mt|metadata|]
+
+metadataJSON :: MetadataConfig -> TZ16.Metadata (ToT Storage)
+metadataJSON MetadataConfig{..} = mconcat
+  [ TZ16.name mcName
+  , TZ16.description mcDescription
+  , TZ16.authors [ TZ16.author "Serokell" "https://serokell.io"
+                 , TZ16.author "Tezos Commons" "https://tezoscommons.org/" ]
+  , TZ16.version versionString
+  , TZ16.license $ TZ16.License "MIT" Nothing
+  , TZ16.homepage "https://github.com/tezos-commons/homebase-lite"
+  , TZ16.errors $(do
+        ClassI _ instances <- reify ''HomebaseLiteError
+        foldr (appE . appE [|(:)|]) [|[]|] $ instances <&> \case
+          InstanceD _ _ (AppT (ConT _) ty@(LitT (StrTyLit _))) _
+            -> [| mkErrorDesc $ Label @($(pure ty)) |]
+          x -> fail $ "Unexpected instance: " <> show x
+      )
```

Comment:
Not a bad system BTW, but it's a bit sad that this is even needed.
`morley-metadata` could have better support for this field perhaps.

Reply:
Perhaps, but I can't come up with robust enough options to generalize this, frankly. One obvious option is to just enumerate error labels (probably as strings, otherwise too much boilerplate) and pass those to some generator function, but it's barely an upgrade over manually calling something like `mkErrorDesc` on each label.

The primary drawback of this approach is it implicitly assumes all error instances are in scope. There's no way to guarantee that, unfortunately. But if we can tolerate this drawback, this TH snippet can be moved to `morley-metadata` with minimal changes.

Reply:
I suppose that it should be possible to grab this info from the autodoc, maybe?

As all the used errors are defined there it should be possible to get the data needed to fill this field.

Note that there is also [morley-metadata#4](https://gitlab.com/morley-framework/morley-metadata/-/issues/4) already that aims at doing this (not only for errors), but first intends to pt TZIP-16 docs in the autodoc to begin with.

Reply:
> I suppose that it should be possible to grab this info from the autodoc, maybe?

That may be doable, actually. Let me see what I can cook up.

Reply:
Yep, that works surprisingly well. See 95dbb3a, `collectContractErrors`. I had to do some reshuffling to avoid circular module imports, though.

Reply:
Indeed, that seems to have worked just fine :ok_hand: